packages/vue_frontend_rpc/test/backend-mock.js

```diff
@@ -0,0 +1,30 @@
+const ASYNC_NOOP = async () => {};
+
+const DefaultRPCFunctions = {
+  onFrontendReady: ASYNC_NOOP,
+  save: ASYNC_NOOP,
+};
+
+/**
+ * This function gets some function implementations that should be available when calling via RPC.
+ * @param userProvidedRpcApis - functions implementations
+ * @returns invoke mock function. When it will be called with the injected function implementation, this given mock implementation
+ * will be executed. Other non given functions will have default empty implementation.
+ */
+export function backendMock(userProvidedRpcApis) {
+  return async function invokeMock(funcName, params) {
+    const actualOpts = {};
+    for (const [rpcFuncName, rpcFuncDefault] of Object.entries(
+      DefaultRPCFunctions
+    )) {
+      // choose between user (optional) provided and default (NOOP)
+      actualOpts[rpcFuncName] =
+        userProvidedRpcApis[rpcFuncName] || rpcFuncDefault;
+    }
+    if (actualOpts[funcName]) {
```

Comment:
why do we build the `actualOpts`in every single invocation of `invokeMock`?
Would the actualOpts change if we call `invokeMock` once or a dozen times?

It feels as though this combines both config/options handling and runtime "engine" concerns.

Reply:
you are right. I fixed it - I put the setup of `actualOps` outside the logic of `invokeMock`